hippod/cache_achievements.py

```diff
@@ -32,3 +101,80 @@ def update(self, achievement):
         if category not in self.data['achievement-results-by-category']:
             self.data['achievement-results-by-category'][category] = list()
         self.data['achievement-results-by-category'][category].append(result)
+
+        # save here results for sunburn data structre which is special
+        # with keys like name, children, size
+        # children-parent-hierarchy based on categories levels e.g. team->topic-subtopic
+        buffer_stat = self.data['achievement-results-sunburn-chart']
+        cat = eval(category)
+        root = cat[0]
+        for j in range(0,len(cat)):
```

Comment:
ok, all if elif else blocks should be moved into separate functions, including comments, it is really hard what is achieved here. Too much code in one method